recipe/thirdparty/api/utils.go

```diff
@@ -1,48 +1,80 @@
 package api
 
-import "github.com/supertokens/supertokens-golang/recipe/thirdparty/tpmodels"
+import (
+	"errors"
 
-func findRightProvider(providers []tpmodels.TypeProvider, thirdPartyId string, clientId *string) *tpmodels.TypeProvider {
+	"github.com/supertokens/supertokens-golang/recipe/thirdparty/tpmodels"
+	"github.com/supertokens/supertokens-golang/supertokens"
+)
 
-	/* logic to find the right provider
-
-	ClientID not passed
-		Case 1: Single Config - Return the config
-		Case 2: Multiple Config - Return the config with IsDefault = true
+func findProvider(options tpmodels.APIOptions, thirdPartyId string, tenantId *string) (tpmodels.TypeProvider, error) {
```

Comment:
change this to:
- if tenant is nil, return from provider array
- if tenant is not nil
   - if provider array has thirdPartyId, return that; else
   - just return a new provider with the thirdPartyId and "empty" configs - this should not require a query the core.

Reply:
Done